File: programmers/코딩테스트 고득점 Kit/해시/전화번호 목록.cpp
#include <string>
#include <vector>
#include <algorithm>
#include <map>
using namespace std;
map <string, int> m;
bool solution(vector<string> phone_book) {
	for (int i = 0; i < phone_book.size(); i++)
		m[phone_book[i]] = 1;
	for (int i = 0; i < phone_book.size(); i++) {
		for (int j = 0; j < phone_book[i].size() - 1; j++) {
			string com = phone_book[i].substr(0, j + 1);
			if (m[com]) return false;
		}
	}
	return true;
}

File: programmers/코딩테스트 고득점 Kit/해시/전화번호 목록.java
import java.util.;
class Solution {
    public boolean solution(String[] phone_book) {
        boolean answer = true;
        HashMapString, Integer m = new HashMap();
        for(String num  phone_book)
            m.put(num, 1);
        for(String num  phone_book)
            for(int i = 0; i  < num.length() - 1; i++) {
                String com = num.substring(0,i+1);
                if(m.containsKey(com)) answer = false;
            }
        return answer;
    }
}